dump/pldm_utils.cpp

```diff
@@ -1,93 +1,174 @@
+                   "to initPLDMInstanceIdDb failed"));
+    }
+}
 
-    method.append(path);
-    method.append(std::vector<std::string>({interface}));
+void destroyPLDMInstanceIdDb()
+{
+    auto rc = pldm_instance_db_destroy(pldmInstanceIdDb);
+    if (rc)
+    {
+        log<level::ERR>(
+            std::format("pldm_instance_db_destroy failed RC{}", rc).c_str());
+    }
+}
 
-    std::vector<std::pair<std::string, std::vector<std::string>>> response;
+pldm_instance_id_t getPLDMInstanceID(uint8_t tid)
+{
+    pldm_instance_id_t instanceID = 0;
 
-    try
+    auto rc = pldm_instance_id_alloc(pldmInstanceIdDb, tid, &instanceID);
+    if (rc == -EAGAIN)
     {
-        auto reply = bus.call(method);
-        reply.read(response);
-        if (response.empty())
-        {
-            log<level::ERR>(fmt::format("Error in mapper response for getting "
-                                        "service name, PATH({}), INTERFACE({})",
-                                        path, interface)
-                                .c_str());
-            return std::string{};
-        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        rc = pldm_instance_id_alloc(pldmInstanceIdDb, tid, &instanceID);
     }
-    catch (const sdbusplus::exception::exception& e)
+
+    if (rc)
     {
-        log<level::ERR>(fmt::format("Error in mapper method call, "
-                                    "errormsg({}), PATH({}), INTERFACE({})",
-                                    e.what(), path, interface)
-                            .c_str());
-        return std::string{};
+        log<level::ERR>(
+            std::format(
+                "getPLDMInstanceID: Failed to alloc ID for TID {}. RC{}", tid,
+                rc)
+                .c_str());
+        elog<NotAllowed>(
+            Reason("Failure in communicating with libpldm service, "
+                   "service may not be running"));
     }
-    return response[0].first;
+    log<level::INFO>(
+        std::format("got tid {} and set instanceID to {}", tid, instanceID)
+            .c_str());
+
+    return instanceID;
 }
-} // namespace internal
 
-using NotAllowed = sdbusplus::xyz::openbmc_project::Common::Error::NotAllowed;
-using Reason = xyz::openbmc_project::Common::NotAllowed::REASON;
+void freePLDMInstanceID(pldm_instance_id_t instanceID, uint8_t tid)
+{
+    auto rc = pldm_instance_id_free(pldmInstanceIdDb, tid, instanceID);
+    if (rc)
+    {
+        log<level::ERR>(
+            std::format(
+                "freePLDMInstanceID: Failed to free ID {} for TID {}. RC{}",
+                instanceID, tid, rc)
+                .c_str());
+    }
+}
 
 int openPLDM()
 {
-    auto fd = pldm_open();
+    auto fd = -1;
+    if (pldmTransport)
+    {
+        log<level::ERR>("openPLDM: pldmTransport already setup!");
+        elog<NotAllowed>(
+            Reason("Required host dump action via pldm is not allowed because "
+                   "pldmTransport already setup"));
```

Comment:
this code will throw exception and return from this method, i guess this is not your intention